### DISCUSSION TITLE
What's the front face?

### DISCUSSION THREAD

**Discussion by jwwalker** [google/filament#4146] [Q&A] [answered]
`MaterialInstance::setCullingMode` determines whether front or back faces are culled, but how is "front" defined?  Is there anything like the OpenGL `glFrontFace` call?

**Comment by prideout** [comment, marked as answer]
It's like default OpenGL -- front faces are primitives that have counter-clockwise winding in screenspace.  Additionally, we automatically flip the directionality when the determinant of the transform matrix is negative to support mirroring.

**Reply by jwwalker** [reply]
Great, handling orientation-reversing transformations is probably the only reason I would need it.
